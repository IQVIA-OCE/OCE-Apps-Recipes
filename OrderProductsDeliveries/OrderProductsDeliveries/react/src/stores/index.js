import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import ordersListReducers from './orders';
import notificationReducer from './notification';

const reducer = combineReducers({
    ordersListReducers,
    notification: notificationReducer,
});

const store = configureStore({ reducer });

export default store;
