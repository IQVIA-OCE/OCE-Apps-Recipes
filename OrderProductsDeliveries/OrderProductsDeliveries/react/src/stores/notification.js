import { createSlice } from '@reduxjs/toolkit';

export const initialState = {
  variant: '',
  text: '',
  icon: '',
  visible: false
};

const notificationSlice = (_initialState) => createSlice({
  name: 'notification',
  initialState: _initialState,
  reducers: {
    showErrorNotification: (state, action) => {
      state.variant = 'error';
      state.icon = 'alert-circle';
      state.text = action.payload;
      state.visible = true;
    },
    closeNotification: (state, action) => {
      state.visible = false;
    },
  }
});

export const notification = notificationSlice(initialState);

export const { showErrorNotification, closeNotification } =
    notification.actions;

export const notificationReducer = notification.reducer;