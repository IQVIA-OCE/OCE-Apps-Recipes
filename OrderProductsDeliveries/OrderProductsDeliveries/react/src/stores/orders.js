import { createSlice } from '@reduxjs/toolkit';
import { fetchOrderById, fetchAllOrderDetailsRecords } from '../api/OrderDetails';
import { normalizeOrderDeliveries, extractBrandsOptions } from '../utils/utils';
import { RecordType } from '../constants';
import { Keyboard } from 'react-native';

export const initialState = {
  orders: [],
  loading: false,
  error: false,
  _recordId: null,
  _recordType: null,
  searchProductValue: null,
  brandOptions: [],
  brandFilter: null
};

const slice = createSlice({
  name: 'ordersList',
  initialState,
  reducers: {
    setOrderDeliveries: (state, action) =>
      Object.assign({}, state, {
        orders: action.payload,
      }),
    setRecordId: (state, action) =>
      Object.assign({}, state, {
        _recordId: action.payload,
      }),
    setRecordType: (state, action) =>
      Object.assign({}, state, {
        _recordType: action.payload,
      }),
    setLoading: (state, action) =>
      Object.assign({}, state, {
        loading: action.payload,
      }),
    setSearchProductValue: (state, action) =>
      Object.assign({}, state, {
        searchProductValue: action.payload,
      }),
    setBrandFilterValue: (state, action) =>
      Object.assign({}, state, {
        brandFilter: action.payload,
      }),
    setBrandOptions: (state, action) =>
      Object.assign({}, state, {
        brandOptions: action.payload,
      }),
  },
});

export default slice.reducer;

export const {
  setOrderDeliveries,
  setRecordId,
  setRecordType,
  setSearchProductValue,
  setLoading,
  setBrandOptions,
  setBrandFilterValue
} = slice.actions;

export function fetchOrderDetailsAsync(recordType, recordId) {
  return async function (dispatch, getState) {
    const { ordersListReducers } = getState();
    const { searchProductValue, brandFilter, brandOptions } = ordersListReducers;

    try {
      const days = 30;

      dispatch(setLoading(true));
      const productDetails = await fetchAllOrderDetailsRecords(
        recordId,
        recordType,
        days,
        searchProductValue,
        brandFilter?.value || null
      );
      const orderDeliveriesByProduct = normalizeOrderDeliveries(productDetails);
      if (brandFilter) {
        const isNoneExist = brandOptions.some(el => el.value === 'none');
        dispatch(setBrandOptions(isNoneExist ? brandOptions : [{ label: 'None', value: 'none' }, ...brandOptions]))
      } else {
        const brandsOptions = extractBrandsOptions(orderDeliveriesByProduct);
        dispatch(setBrandOptions(brandsOptions))
      }
      dispatch(setOrderDeliveries(orderDeliveriesByProduct));
      dispatch(setLoading(false));
      Keyboard.dismiss();
    } catch (err) {
      console.log(err);
    }
  };
}

export function setRecordInfoAsync(recordId) {
  return async function (dispatch) {
    try {
      dispatch(setLoading(true));
      const ordersResponse = await fetchOrderById(recordId);
      if (ordersResponse.records.length) {
        dispatch(setRecordType(RecordType.Order));
      } else {
        dispatch(setRecordType(RecordType.Account));
      }
      dispatch(setRecordId(recordId));
      dispatch(setLoading(false));
    } catch (err) {
      console.log(err);
    }
  };
}
