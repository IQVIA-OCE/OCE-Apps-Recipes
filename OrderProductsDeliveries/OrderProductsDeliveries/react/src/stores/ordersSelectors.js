export const ordersSelector = (state) => state.orders.orders;
export const loadingSelector = (state) => state.orders.loading;
export const errorSelector = (state) => state.orders.error;
export const _recordIdSelector = (state) => state.orders._recordId;
export const searchProductValueSelector = (state) => state.orders.searchProductValue;
export const brandOptionsSelector = (state) => state.orders.brandOptions;
export const brandFilterSelector = (state) => state.orders.brandFilter;
export const isAccountFilterVisibleSelector = (state) => state.orders.isAccountFilterVisible;
