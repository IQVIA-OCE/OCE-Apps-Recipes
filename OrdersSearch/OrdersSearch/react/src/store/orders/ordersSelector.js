export const ordersListSelector = state => state.orders.ordersList.list;
export const ordersListLoadingSelector = state => state.orders.ordersList.loadingStatus;
export const recordIdSelector = state => state.orders._recordId;
export const accountIdSelector = state => state.orders.account.id;
export const accountNameSelector = state => state.orders.account.name;
export const accountNameLoadingSelector = state => state.orders.account.loadingStatus;
export const pageTypeSelector = state => state.orders.pageType;
export const errorSelector = state => state.orders.error;