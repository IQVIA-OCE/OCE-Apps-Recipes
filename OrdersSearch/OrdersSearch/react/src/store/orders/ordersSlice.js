import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { fetchAccount, fetchAllOrders } from "../../api/ordersApi";
import { LOADING_STATUS } from "../../utils/constants";

export const initialState = {
  ordersList: {
    list: [],
    loadingStatus: LOADING_STATUS.IDLE,
  },
  account: {
    name: null,
    id: null,
    loadingStatus: LOADING_STATUS.IDLE,
  },
  pageType: null,
  error: null 
};

//0016t00000U7Df6AAF  --- account

//a3s6t000000KzqzAAC  - order

export const fetchOrdersList = createAsyncThunk(
  "orders/fetchOrdersList",
  async (id, { getState, rejectWithValue }) => {
    try {
      const { filters, orders } = getState();
      const records = await fetchAllOrders(filters.values, id || orders.account.id);

      return records;
    } catch (error) {
      return rejectWithValue(error);
    }
  }
);

export const fetchAccountName = createAsyncThunk(
  "orders/fetchAccountName",
  async (recordId, { getState, rejectWithValue }) => {
    try {
      const { records } = await fetchAccount(recordId);
      return records[0].Name;
    } catch (error) {
      return rejectWithValue(error);
    }
  }
);

const ordersSlice = createSlice({
  name: "orders",
  initialState,
  reducers: {
    setAccountId: (state, { payload }) => {
      state.account.id = payload;
    },
    setAccountName: (state, { payload }) => {
      state.account.name = payload;
    },
    setPageType: (state, { payload }) => {
        state.pageType = payload;
    },
    setError: (state, { payload }) => {
      state.error = payload;
    }
  },
  extraReducers: {
    [fetchOrdersList.fulfilled]: (state, action) => {
      state.ordersList.loadingStatus = LOADING_STATUS.SUCCESS;
      state.ordersList.list = action.payload;
    },
    [fetchOrdersList.pending]: (state) => {
      state.ordersList.loadingStatus = LOADING_STATUS.PENDING;
      state.error = null;
    },
    [fetchOrdersList.rejected]: (state, action) => {
      state.ordersList.loadingStatus = LOADING_STATUS.FAILED;
      state.error = action.payload;
    },
    [fetchAccountName.fulfilled]: (state, action) => {
      state.account.loadingStatus = LOADING_STATUS.SUCCESS;
      state.account.name = action.payload;
    },
    [fetchAccountName.pending]: (state) => {
      state.account.loadingStatus = LOADING_STATUS.PENDING;
      state.error = null;
    },
    [fetchAccountName.rejected]: (state, action) => {
      state.account.loadingStatus = LOADING_STATUS.FAILED;
      state.error = action.payload;
    },
  },
});

export const ordersSliceReducer = ordersSlice.reducer;

export const { setAccountId, setAccountName, setPageType, setError } = ordersSlice.actions;
