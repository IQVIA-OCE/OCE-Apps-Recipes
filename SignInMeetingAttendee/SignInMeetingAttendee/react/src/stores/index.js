import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import { meeting } from './meeting';

const reducer = combineReducers({ meeting });

const store = configureStore({ reducer });

export default store;
