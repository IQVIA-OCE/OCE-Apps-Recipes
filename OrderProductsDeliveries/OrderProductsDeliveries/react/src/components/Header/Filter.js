import React from "react";
import { View, StyleSheet } from "react-native";
import { Select } from 'apollo-react-native';
import { useSelector, useDispatch } from "react-redux";
import { setBrandFilterValue } from "../../stores/orders";
import {brandFilterSelector, brandOptionsSelector} from "../../stores/ordersSelectors";

export default () => {

    const dispatch = useDispatch();
    const brandsOptions = useSelector(brandOptionsSelector);
    const brandFilterValue = useSelector(brandFilterSelector);

    return (
        <View style={styles.content} testID="filterWrapper">
            <Select
                options={brandsOptions}
                value={brandFilterValue}
                onChange={val => dispatch(setBrandFilterValue(val?.value === 'none' ? null : val))}
                placeholder="Brand"
                width={'100%'}
                disabled={brandsOptions.length ? false : true}
                hideDropdownPlaceholder={true}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    content: {
        width: 300
    },
});
