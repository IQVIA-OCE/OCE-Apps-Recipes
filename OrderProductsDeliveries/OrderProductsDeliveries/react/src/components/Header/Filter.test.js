import { render } from '@testing-library/react-native';
import React from 'react';
import Filter from './Filter';
import { useSelector, useDispatch } from 'react-redux';
import { Select } from 'apollo-react-native';
import { setBrandFilterValue } from "../../stores/orders";
// import renderer from 'react-test-renderer';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock('../../stores/orders', () => ({
    setBrandFilterValue: jest.fn()
}));

describe('Filter', () => {
    it('without data', () => {
        useSelector.mockImplementation((cb) => cb({
            ordersListReducers: {
                brandOptions: [],
                brandFilter: ''
            }
        }));
        const { container, toJSON } = render(<Filter />);
        expect(toJSON(container)).toMatchSnapshot();
    })
    it('with data', () => {
        useDispatch.mockReturnValue(() => jest.fn());
        useSelector.mockImplementation((cb) => cb({
            ordersListReducers: {
                brandOptions: [{ label: '123', id: '123'}],
                brandFilter: '12'
            }
        })); 
        const { container, toJSON } = render(<Filter />);
        expect(toJSON(container)).toMatchSnapshot();
        
        container.findAllByType(Select)[0].props.onChange({ value: 'none' });
        expect(setBrandFilterValue).toBeCalledTimes(1);
    })

})