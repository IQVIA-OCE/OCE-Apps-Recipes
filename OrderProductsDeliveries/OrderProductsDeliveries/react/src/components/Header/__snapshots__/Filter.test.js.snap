// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter with data 1`] = `
<View
  style={
    Object {
      "width": 300,
    }
  }
  testID="filterWrapper"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": undefined,
                "borderColor": "#d9d9d9",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    false,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Filter without data 1`] = `
<View
  style={
    Object {
      "width": 300,
    }
  }
  testID="filterWrapper"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "rgba(246, 246, 246, 0.4)",
                "borderColor": "rgba(217, 217, 217, 0.4)",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.4)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                Brand
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    Object {
                      "opacity": 0.32,
                    },
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.4)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
